proposals/trusted_registry_crd.md

```diff
@@ -0,0 +1,128 @@
+  report associated with a given container image digest, for example `registryvulnerabilities.aquasecurity.github.io`.
+- Container images are not first-class citizens in Kubernetes-native world. There’s no built-in resource that represents
+  a container image. Also, there’s no built-in representation of a (trusted) registry.
+- An admission controller has certain performance requirements. Usually we cannot trigger a scanner when the admission
+  controller is invoked for a given Kubernetes API request. Most likely it will take way too long to perform such
+  ad hoc scans.
+
+## Solutions
+
+### Define registries.aquasecurity.github.io resource
+
+First off, introduce a custom resource to represent a (trusted) container registry. For example, we can represent
+a private Harbor registry with the following Kubernetes object:
+
+```yaml
+apiVersion: aquasecurity.github.io/v1alpha1
+kind: Registry
+metadata:
+  name: my-harbor-registry
+spec:
+  name: My Harbor Registry
+  description: |
+    That's the only registry authorized to run production workloads.
+  server: core.harbor.domain
+  imagePullSecrets:
+  - name: my-harbor-registry-credentials
+  insecure: false
+  trustedRepositories:
+  - name: library/nginx:1.16
+  - name: library/alpine
+  - namePattern: trusted/**
+status:
+  trustedRepositoryStatuses:
+  - name: library/nginx
+    scanned: true
+    lastScanTime: 2020-01-01T00:00:00Z
+  - name: library/alpine
+    scanned: false
+    lastScanTime: null
+```
+
+The `Registry` entity has the `spec` property, which allows specifying trusted repositories. The `status` filed might
+contain scan status for individual repositories. In the example above, you can see that the `library/nginx` repository
+has already been scanned, whereas the `library/alpine` repository has not.
+
+### Define registryvulnerabilities.aquasecurity.github.io resource
+
+Define the `RegistryVulnerability` entity to represent a vulnerability report for the specified image digest:
+
+
+```yaml
+apiVersion: aquasecurity.github.io/v1alpha1
+kind: RegistryVulnerability
+metadata:
+  name: ddbb8bfb-040a-42a1-a051-d5aff77a53e7
+  labels:
+    starboard.repository.name: library/nginx
+    starboard.repository.tag: 1.16
+    starboard.repository.digest: sha256.d20aa6d1cae56fd17cd458f4807e0de462caf2336f0b70b5eeb69fcaaf30dd9c
+    starboard.resource.kind: Registry
+    starboard.resource.name: my-harbor-registry
+report:
+  scanner:
+    name: Trivy
+    vendor: Aqua Security
+    version: 0.9.1
+  registry:
+    url: https://core.harbor.domain
+  artifact:
+      repository: "library/nginx"
+      digest: "sha256:d20aa6d1cae56fd17cd458f4807e0de462caf2336f0b70b5eeb69fcaaf30dd9c"
+      tag: "1.16"
+      mimeType: "application/vnd.docker.distribution.manifest.v2+json"
+  summary:
+    criticalCount: 0
+    highCount: 0
+    lowCount: 92
+    mediumCount: 27
+    noneCount: 0
+    unknownCount: 3
+  vulnerabilities:
+  - id: CVE-2020-1879
```

Comment:
Did you just omit details such as a package name and CVSS? They are necessary so that a complicated policy can be applied in Admission Controller.

Reply:
We'll have all the details as we have for `vulnerabilities.aquasecurity.github.io`. For example, https://github.com/aquasecurity/starboard/blob/master/kube/examples/vulnerabilities.yaml#L30 I've update the sample snippet.